integrationTesting/tests/organize/surveys/submitting-survey.spec.ts

```diff
@@ -39,6 +40,84 @@ test.describe('User submitting a survey', () => {
     moxy.teardown();
   });
 
+  test('submits text input', async ({ appUri, moxy, page }) => {
+    await page.goto(
+      `${appUri}/o/${KPDMembershipSurvey.organization.id}/surveys/${KPDMembershipSurvey.id}`
+    );
+
+    await page.fill('[name="2.text"]', 'Topple capitalism');
+    await page.click('input[name="sig"][value="anonymous"]');
+    await page.click('data-testid=Survey-acceptTerms');
+    await Promise.all([
+      page.waitForResponse((res) => res.request().method() == 'POST'),
+      await page.click('data-testid=Survey-submit'),
+    ]);
+
+    const log = moxy.log(`/v1${apiPostPath}`);
+    expect(log.length).toBe(1);
+    const [request] = log;
+    const data = request.data as {
+      responses: ZetkinSurveyQuestionResponse[];
+    };
+    expect(data.responses).toMatchObject([
+      {
+        question_id: KPDMembershipSurvey.elements[1].id,
+        response: 'Topple capitalism',
+      },
+    ]);
+  });
+
+  test('required text input blocks submission when empty', async ({
+    appUri,
+    moxy,
+    page,
+  }) => {
+    moxy.setZetkinApiMock(
+      `/orgs/${KPDMembershipSurvey.organization.id}/surveys/${KPDMembershipSurvey.id}`,
+      'get',
+      {
+        ...KPDMembershipSurvey,
+        elements: KPDMembershipSurvey.elements.map((element, i) => {
+          if (i === 1) {
```

Comment:
I worry about this approach a little, because it relies on the `KPDMembershipSurvey` fixture to not change, which it shouldn't, but sometimes we do change them.

Maybe you could just replace `elements` with a single-item array with a hard-coded text field here? That way, the test is more readable (because the full fake data is in the test) and also doesn't rely as heavily on the `KPDMembershipSurvey` fake to stay the same forever?

Reply:
oh nice I was hoping for a comment along these lines, drew a total blank when trying to figure out how to do this in a non ugly way and figured you'd have an idea

Reply:
Glad to help! Just to clarify, I still think using the KPD fake makes sense for the outer (non-element) data, since that isn't really subject to the test and just needs to be there for the object to have the correct shape.